#! /bin/sh -e

# DP: Backport fix for PR libmudflap/26864 from the trunk.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

2006-04-22  Frank Ch. Eigler  <fche@redhat.com>

	PR libmudflap/26864
	* common.opt (flag_mudflap_threads): Overload flag_mudflap.
	* tree-mudflap.c (flag_mudflap_threads): Derive from flag_mudflap.
 
Index: gcc/tree-mudflap.c
===================================================================
--- gcc/tree-mudflap.c	(revision 113178)
+++ gcc/tree-mudflap.c	(revision 113179)
@@ -48,6 +48,10 @@
 
 /* Internal function decls */
 
+
+/* Options.  */
+#define flag_mudflap_threads (flag_mudflap == 2)
+
 /* Helpers.  */
 static tree mf_build_string (const char *string);
 static tree mf_varname_tree (tree);
Index: gcc/common.opt
===================================================================
--- gcc/common.opt	(revision 113178)
+++ gcc/common.opt	(revision 113179)
@@ -605,7 +605,7 @@
 Add mudflap bounds-checking instrumentation for single-threaded program
 
 fmudflapth
-Common RejectNegative Report Var(flag_mudflap_threads)
+Common RejectNegative Report VarExists Var(flag_mudflap,2)
 Add mudflap bounds-checking instrumentation for multi-threaded program
 
 fmudflapir
